src/main/webapp/app/example-text-submission/example-text-submission.component.ts

```diff
@@ -0,0 +1,289 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Location } from '@angular/common';
+import { JhiAlertService } from 'ng-jhipster';
+import { Subscription } from 'rxjs';
+import { AccountService } from '../core';
+import { ExampleSubmission } from 'app/entities/example-submission';
+import { ExerciseService } from 'app/entities/exercise';
+import { HttpResponse, HttpErrorResponse } from '@angular/common/http';
+import { TextSubmission, TextSubmissionService } from 'app/entities/text-submission';
+import { ExampleSubmissionService } from 'app/entities/example-submission/example-submission.service';
+import { Feedback } from 'app/entities/feedback';
+import { TextAssessmentsService } from 'app/entities/text-assessments/text-assessments.service';
+import { Result } from 'app/entities/result';
+import { HighlightColors } from 'app/text-shared/highlight-colors';
+import { TextExercise } from 'app/entities/text-exercise';
+import { TutorParticipationService } from 'app/tutor-exercise-dashboard/tutor-participation.service';
+import { TutorParticipation } from 'app/entities/tutor-participation';
+
+@Component({
+    selector: 'jhi-example-text-submission',
+    templateUrl: './example-text-submission.component.html',
+    providers: [JhiAlertService]
+})
+export class ExampleTextSubmissionComponent implements OnInit {
+    isNewSubmission: boolean;
+    areNewAssessments = true;
+    exerciseId: number;
+    exampleSubmission = new ExampleSubmission();
+    textSubmission = new TextSubmission();
+    assessments: Feedback[] = [];
+    assessmentsAreValid = false;
+    result: Result;
+    totalScore: number;
+    invalidError: string;
+    exercise: TextExercise;
+    isAtLeastInstructor = false;
+    readOnly: boolean;
+    toComplete: boolean;
+
+    public getColorForIndex = HighlightColors.forIndex;
+
+    private exampleSubmissionId: number;
+    private subscription: Subscription;
+
+    constructor(
+        private exerciseService: ExerciseService,
+        private textSubmissionService: TextSubmissionService,
+        private exampleSubmissionService: ExampleSubmissionService,
+        private assessmentsService: TextAssessmentsService,
+        private tutorParticipationService: TutorParticipationService,
+        private jhiAlertService: JhiAlertService,
+        private accountService: AccountService,
+        private route: ActivatedRoute,
+        private router: Router,
+        private location: Location
+    ) {
+    }
+
+    ngOnInit(): void {
+        // (+) converts string 'id' to a number
+        this.subscription = this.route.params.subscribe(params => {
```

Comment:
Subscription is never unsubscribed.
You can also use route snapshot if you do not plan to listen on route changes:
https://github.com/ls1intum/ArTEMiS/blob/develop/src/main/webapp/app/text-assessment/text-assessment.component.ts#L50